src/arrayMethodSort.js

```diff
@@ -4,8 +4,27 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction) {
+    let callBack = compareFunction;
+
+    if (!callBack) {
+      callBack = function (a, b) {
+        return String(a) > String(b) ? 1 : -1;
+      };
+    }
+
+    for (let i = 0; i < this.length - 1; i++) {
+      for (let j = i + 1; j < this.length; j++) {
+        if (callBack(this[i], this[j]) > 0) {
+          const backup = this[i];
+
+          this[i] = this[j];
+          this[j] = backup;
+        }
+      }
+    }
```

Comment:
You are using a bubble sort algorithm here, which is fine for this task. But keep in mind that bubble sort has a worst-case and average complexity of O(n^2), where n is the number of items being sorted. Therefore, it's not suitable for large arrays. As an improvement, you might want to consider using a more efficient sorting algorithm such as quicksort or mergesort for larger arrays.